### ISSUE TITLE
Values not selected on first render

### ISSUE THREAD

**Issue by landy2005** [mistic100/jQuery-QueryBuilder#661]
Here partial script/values example
```
  var rules = {"condition":"AND","rules":[{"id":"device.device_id","field":"device.device_id","type":"string","input":"select","operator":"in","value":["275","166","165"]}],"valid":true};

  $('#rules-nqqx74gQ').queryBuilder({
    plugins: {
      'bt-selectpicker': null,
      'sortable': null,
    },
    filters:  [ 
{ id: 'device.device_id',
field: 'device.device_id',
label: 'Device',
type: 'string',
input: 'select',
plugin: 'selectpicker',
values:  [ { '275': '109.74.240.30' }, { '186': '123.176.37.12' }, { '251': '141.255.191.124' }, { '269': '145.253.90.172' }, { '203': '151.79.240.212' }, { '278': '179.97.83.118' }, { '193': '185.12.128.1' }, { '246': '202.188.31.178' }, { '218': '206.172.16.198' }, { '207': '208.124.112.60' }, { '201': '209.160.135.153' }, { '267': '210.119.219.157' }, { '210': '5.148.148.178' }, { '273': '58.65.247.24' }, { '217': '66.103.255.102' }, { '219': '74.143.95.21' }, { '184': '80.156.183.198' }, { '188': '83.69.220.154' }, { '241': '94.199.173.205' }, { '59': 'aix-1.test' }, { '121': 'aix-2.test' }, { '105': 'allied1.test' }, { '106': 'allied2.test' }, { '58': 'aos.test' }, { '56': 'aos7.test' }, { '22': 'apc-8953.test' }, { '33': 'apc-ap7822.test' }, { '32': 'apc-ap7862.test' }, { '31': 'apc-ap8941.test' }, { '12': 'apc-ap9340-2.test' }, { '11': 'apc-ap9340.test' }, { '113': 'apc-ap9631.test' }, { '215': 'apc-netbotz.test' }, { '271': 'apc-netbotz2.test' }, { '249': 'apc-symmetra.test' }, { '147': 'arista.test' }, { '50': 'aruba1.test' }, { '116': 'aten-pdu.test' }, { '144': 'avaya-server.test' }, { '42': 'avaya.test' }, { '142': 'baytech.test' }, { '183': 'bintec1.test' }, { '67': 'breeze1.test' }, { '68': 'breeze2.test' }, { '69': 'breeze3.test' }, { '71': 'breeze4.test' }, { '72': 'breeze5.test' }, { '70': 'breezemax.test' }, { '237': 'brocade-bgp.test' }, { '99': 'brocade-fws.test' }, { '235': 'brocade-mlxe.test' }, { '140': 'c01-jbt1-osl-no.nexthop.no' }, { '17': 'calix-c7.test' }, { '168': 'casa-100g.test' }, { '146': 'casa.test' }, { '110': 'cc-plus.test' }, { '77': 'checkpoint.test' }, { '238': 'chloride.test' }, { '103': 'cisco-12k.test' }, { '96': 'cisco-2960c.test' }, { '114': 'cisco-3800.test' }, { '93': 'cisco-7600-pw.test' }, { '94': 'cisco-7600-pw2.test' }, { '136': 'cisco-ace.test' }, { '143': 'cisco-altiga.test' }, { '209': 'cisco-asa.test' }, { '92': 'cisco-asr9k-pw.test' }, { '141': 'cisco-bng.test' }, { '166': 'cisco-catos.test' }, { '236': 'cisco-mac.test' }, { '194': 'cisco-nexus5.test' }, { '208': 'cisco-nexus9.test' }, { '167': 'cisco-nxos-ports.test' }, { '213': 'cisco-nxos-vrf.test' }, { '129': 'cisco-nxos.test' }, { '81': 'cisco-ons.test' }, { '135': 'cisco-spa.test' }, { '131': 'cisco-ubr.test' }, { '250': 'cisco-ucs.test' }, { '88': 'cisco-wlc.test' }, { '14': 'ciscosb-2.test' }, { '162': 'ciscosb-4.test' }, { '163': 'ciscosb-dom.test' }, { '229': 'ciscosb-poe.test' }, { '10': 'ciscosb.test' }, { '239': 'cs141.test' }, { '19': 'darwin-i386.test' }, { '20': 'darwin-ppc.test' }, { '1': 'demo.observium.ru' }, { '97': 'dev.observium.ru' }, { '170': 'develop-ineo.test' }, { '108': 'dgs1.test' }, { '109': 'dgs2.test' }, { '122': 'dlink-fw.test' }, { '123': 'dlink-router.test' }, { '111': 'dlink1.test' }, { '112': 'dlink2.test' }, { '192': 'draytek-adsl.test' }, { '280': 'edgeswitch-xp.test' }, { '270': 'eltek-smart.test' }, { '164': 'eltex-10g.test' }, { '221': 'eltex-mes.test' }, { '240': 'enlogic-pdu.test' }, { '227': 'enterasys-poe.test' }, { '54': 'extendair1.test' }, { '53': 'extendair2.test' }, { '232': 'extreme-poe.test' }, { '181': 'extreme-x670.test' }, { '254': 'f5-ha.test' }, { '159': 'f5-ports.test' }, { '91': 'f5.test' }, { '127': 'fabos-sfp.test' }, { '176': 'fastpath.test' }, { '230': 'foundry-poe.test' }, { '55': 'freebsd10.test' }, { '196': 'ftos-dell.test' }, { '204': 'ftos-dell2.test' }, { '36': 'ftos-s.test' }, { '195': 'ftos-s6000.test' }, { '212': 'ge-ups.test' }, { '276': 'geist-pdu-imd.test' }, { '156': 'geist-watchdog.test' }, { '187': 'gr4-the.clearip.net' }, { '154': 'h3c-sla.test' }, { '155': 'h3c-sla2.test' }, { '225': 'hh3c-poe.test' }, { '171': 'hp-ink.test' }, { '65': 'hps01.luna-it-systeme.de' }, { '21': 'huawei-8505.test' }, { '231': 'huawei-poe.test' }, { '161': 'huawei-vrp1.test' }, { '169': 'hwg-pwr.hwg.cz' }, { '61': 'hwg-ste-plus.hwg.cz' }, { '76': 'hwg-ste.test' }, { '255': 'ibm-bladecenter.paid' }, { '83': 'idrac.test' }, { '160': 'idrac8.test' }, { '51': 'iosxe-4500.test' }, { '256': 'iosxe-asr920.test' }, { '4': 'iosxe-ipv6.test' }, { '5': 'iosxe-ipv6bug.test' }, { '3': 'iosxr-ipv6.test' }, { '197': 'isilon.test' }, { '200': 'juniper-dom.test' }, { '158': 'juniper-m20.test' }, { '117': 'junos-1.test' }, { '118': 'junos-2.test' }, { '149': 'junos-boyard.test' }, { '100': 'junos-ipv6.test' }, { '130': 'junos-pw.test' }, { '150': 'junos-pw2.test' }, { '151': 'junos-pw3.test' }, { '268': 'junos-srx.test' }, { '178': 'kyocera.test' }, { '115': 'liebert-hvac.test' }, { '248': 'liebert-pdu.test' }, { '189': 'liebert-ups.test' }, { '25': 'mediant-esbc.test' }, { '35': 'mediant-mp118-2.test' }, { '34': 'mediant-mp118.test' }, { '26': 'mediant-old.test' }, { '205': 'mge-ups-ex.test' }, { '206': 'mge-ups.test' }, { '6': 'mgm-7600.test' }, { '124': 'mickael.test' }, { '102': 'mitel.test' }, { '29': 'mks.stupalov.ru' }, { '211': 'mrv-optiswitch.test' }, { '24': 'nas.stupalov.ru' }, { '120': 'netapp-cluster.test' }, { '73': 'netapp.test' }, { '226': 'netgear-poe.test' }, { '47': 'netgear1.test' }, { '49': 'netgear3.test' }, { '274': 'netonix.test' }, { '157': 'netping.test' }, { '182': 'netscaler.test' }, { '222': 'nimble.test' }, { '43': 'nortel.test' }, { '191': 'nos.test' }, { '153': 'np.tst.netping.ru' }, { '138': 'obs-ex.atomfried.dyndns.org' }, { '139': 'obs-srx.atomfried.dyndns.org' }, { '13': 'observium.net.gpt.ru' }, { '89': 'olivetti.test' }, { '37': 'omnistack.test' }, { '202': 'onefs8.test' }, { '126': 'oneos5.test' }, { '125': 'oneos9.test' }, { '2': 'openbsd.test' }, { '38': 'openwrt-24.test' }, { '39': 'openwrt-26.test' }, { '145': 'openwrt.test' }, { '30': 'packetflux.test' }, { '90': 'peakflow.test' }, { '64': 'pi.stupalov.ru' }, { '82': 'pica8.test' }, { '253': 'portal.nadlinger.at' }, { '63': 'poseidon.hwg.cz' }, { '173': 'powerconnect-old.test' }, { '174': 'powerconnect-radlan.test' }, { '172': 'powerconnect.test' }, { '98': 'procurve-ap.test' }, { '85': 'procurve-kent.test' }, { '228': 'procurve-poe.test' }, { '84': 'procurve.test' }, { '44': 'procurve3.test' }, { '175': 'proliant.test' }, { '40': 'prtg.observium.ru' }, { '223': 'quanta-ly.test' }, { '104': 'radlan.test' }, { '214': 'raritan-emx.test' }, { '247': 'raritan-env.test' }, { '234': 'raritan-pdu.test' }, { '148': 'rhel7.observium.ru' }, { '177': 'ricoh-c.test' }, { '279': 'rittal-iii.test' }, { '7': 'sentry3.test' }, { '128': 'shopik-adaptec.test' }, { '66': 'shopik-huawei.test' }, { '165': 'shopik-qct.test' }, { '252': 'snr-switch.test' }, { '79': 'sofaware.test' }, { '86': 'sonicwall.test' }, { '245': 'sophos-utm.paid' }, { '134': 'sophos.test' }, { '101': 'spb-off-gw1.net.p8.ru' }, { '198': 'sub10.test' }, { '107': 'sw-1916.net.p8.ru' }, { '272': 'synology-dsm.test' }, { '87': 'timos-sas.test' }, { '57': 'timos.test' }, { '199': 'tripplite-pdu.test' }, { '23': 'ubiquiti.test' }, { '220': 'unifi-switch.test' }, { '152': 'uniping-v3.test' }, { '133': 'vds.coosm.net' }, { '41': 'vmware.test' }, { '224': 'vmware6.test' }, { '137': 'vpn6.clearip.net' }, { '185': 'win.observium.ru' }, { '95': 'windows.test' }, { '9': 'windows2003.test' }, { '233': 'wowza.test' }, { '244': 'xppc-3phase.test' }, { '243': 'xppc-env.test' }, { '190': 'xppc-ups.test' }, { '242': 'xppc.test' }, { '180': 'zhone-malk.test' }, { '179': 'zhone-mxk.test' }, { '216': 'zte-switch.test' }, { '119': 'zxr10.test' } ] ,
multiple: true,
operators: ['in','notin'] } ,   ] ,

      //operators: $.fn.queryBuilder.constructor.DEFAULTS.operators.concat([
      operators: ([
      { type: 'le',		      nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'ge',		      nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'lt',		      nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'gt',		      nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'equals',		  nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'notequals',	nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'match',		  nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'notmatch',	  nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'regexp',  	  nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'notregexp',	nb_inputs: 1, multiple: false, apply_to: ['string'] },
      { type: 'in',		      nb_inputs: 1, multiple: true,  apply_to: ['string'] },
      { type: 'notin',		  nb_inputs: 1, multiple: true,  apply_to: ['string'] }
    ]),
    lang: {
      operators: {
        le:		'less or equal',
        ge:		'greater or equal',
        lt:		'less than',
        gt:		'greater than',
        equals:		'equals',
        notequals:	'not equals',
        match:		'match',
        notmatch:	'not match',
        regexp:		'regexp',
        notregexp:	'not regexp',
        in:		'in',
        notin:		'not in'
      }
    },
    rules: rules
  });
```

here values `"275","166","165"` must be selected on first render (for (multi)select), but this does not happen.

As I see in code, here complete no condition for generate select values as selected (on first render).
I think same issue with other input fields, old "know" value not passed for first render.

**Comment by mistic100** [comment]
Selectpicker does not watch the select value (shame !), you need to call the "render" [method](https://silviomoreto.github.io/bootstrap-select/methods/#selectpickerrender)

```js
$('#rules-nqqx74gQ').on('afterUpdateRuleValue.queryBuilder', function(rule) {
  rule.$el.find('.rule-value-container select').selectpicker('render');
});
```

The event handler be registered **before** QueryBuilder init otherwise it won't work on first render neither.

**Comment by mistic100** [comment]
Because this is done programatically AFTER the creation of the select.

**Comment by landy2005** [comment]
Ok, than this is possible make in code render also AFTER creation of the select?

Currently, your suggestion or not logical, or write this suggestion in the docs.

**Comment by mistic100** [comment]
The only thing that is not logical is Selectpicker not listening to the "change" event on the select, which is a standard DOM event for inputs and selects.

I will not change how that builder works, it is based on standard DOM manipulation and JS API.

**Comment by landy2005** [comment]
Hrm, issue fixed by update selectpicker to v1.13, seems as here added required event.
